opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -588,7 +588,7 @@ def use_span(
         finally:
             context_api.detach(token)
 
-    except BaseException as exc:  # pylint: disable=broad-exception-caught
+    except Exception as exc:  # pylint: disable=broad-exception-caught
```

Comment:
maybe we could add some clarifying notes, e.g. this is what the docs say about GeneratorExit:
> exception GeneratorExit
>    Raised when a [generator](https://docs.python.org/3/glossary.html#term-generator) or [coroutine](https://docs.python.org/3/glossary.html#term-coroutine) is closed; see [generator.close()](https://docs.python.org/3/reference/expressions.html#generator.close) and [coroutine.close()](https://docs.python.org/3/reference/datamodel.html#coroutine.close). It directly inherits from [BaseException](https://docs.python.org/3/library/exceptions.html#BaseException) instead of [Exception](https://docs.python.org/3/library/exceptions.html#Exception) since it is technically not an error.

```suggestion
    # Record only exceptions that inherit Exception class but not BaseException, because 
    # classes that directly inherit BaseException are not technically errors, e.g. GeneratorExit. 
    except Exception as exc:  # pylint: disable=broad-exception-caught
```

Reply:
Also perhaps link the issue in the comments.

Reply:
Added. Thanks